src/care/ProfilePlugin.cpp

```diff
@@ -0,0 +1,66 @@
+#include "care/ProfilePlugin.h"
+
+#include "chai/ArrayManager.hpp"
+
+#include "chai/ExecutionSpaces.hpp"
+
+#include "care/CHAICallback.h"
+
+#include "care/PluginData.h"
+
+/* CUDA profiling macros */
+#if defined(__CUDACC__) && CARE_HAVE_NVTOOLSEXT
+#include "nvToolsExt.h"
+#endif
+
+namespace care{
+   unsigned int ProfilePlugin::s_current_color = 0;
+   uint32_t ProfilePlugin::s_colors[7] = { 0x0000ff00, 0x000000ff, 0x00ffff00, 0x00ff00ff, 0x0000ffff, 0x00ff0000, 0x00ffffff };
+   int ProfilePlugin::s_num_colors = sizeof(s_colors) / sizeof(uint32_t);
+   bool ProfilePlugin::s_profile_host_loops = true; 
+   
+   ProfilePlugin::ProfilePlugin() {}
+
+   void ProfilePlugin::preLaunch(const RAJA::util::PluginContext& p) {
+#if CARE_HAVE_NVTOOLSEXT
+#if defined(CARE_GPUCC)
```

Comment:
This was not your fault - the preexisting code had a bug - but would you be willing to change `CARE_GPUCC` to `__CUDACC__` in ProfilePlugin::preLaunch and ProfilePlugin::postLaunch? `CARE_GPUCC` implies ROCm/Hip support, but NVTX is only supported by CUDA.